File: csharp/largest-series-product/LargestSeriesProduct.cs
using System;
using System.Linq;

public static class LargestSeriesProduct
{
    public static long GetLargestProduct(string digits, int window) =>
        window < 0 || window > digits.Length || digits.Any(w => !char.IsNumber(w))
            ? throw new ArgumentException()
            : Enumerable.Range(0, digits.Length - window + 1)
                .Select(i => digits.Substring(i, window).Select(c => c - '0').Aggregate(1, (x, y) => x * y)).Max();
}

File: vbnet/largest-series-product/LargestSeriesProduct.vb
Imports System

Public Module LargestSeriesProduct
    Public Function GetLargestProduct(ByVal digits As String, ByVal span As Integer) As Long
        If span > digits.Length OrElse span < 0 OrElse digits.Any(Function(c) c > "9"c Or c < "0"c) Then
            Throw New ArgumentException("invalid arguments")
        End If
        Dim ints = digits.Select(Function(c) Convert.ToInt32(c) - Convert.ToInt32("0"c)).ToList()
        Return Enumerable.Range(0, digits.Length - span + 1).Select(Function(i) ints.Skip(i).Take(span)) _
            .Max(Function(chunk) chunk.Aggregate(Function(acc, n) acc * n))
    End Function
End Module
